app/models/katello/docker_meta_tag.rb

```diff
@@ -131,6 +131,7 @@ def self.with_identifiers(ids)
 
     def self.cleanup_tags
       self.where(:schema2_id => nil, :schema1_id => nil).delete_all
+      self.where("id not in (?) OR (schema2_id IS NULL AND schema1_id IS NULL)", Katello::RepositoryDockerMetaTag.pluck(:docker_meta_tag_id)).delete_all
```

Comment:
I think the second half of this query is a duplicate of the line above: schema2_id IS NULL AND schema1_id IS NULL

Reply:
Oh I did mean to remove the above query..let me do that..

Reply:
Updated..